src/Core/SyntaxScope.re

```diff
@@ -0,0 +1,44 @@
+module StringUtil = Utility.StringUtil;
+
+// Type defining a simplified 'syntax scope' of a token -
+// is it a comment, string, both, or neither?
+// This is important for features like auto-closing pairs -
+// we may not care to close certain characters in contexts like strings or comments.
+type t = 
+| None
+| String
+| Comment
+| StringAndComment;
```

Comment:
Maybe I'm missing something, but it seems more appropriate to model this as two `bool`s.  That would greatly simplify the two functions below, and probably other code as well.